pkg/cmd/server/api/validation/master.go

```diff
@@ -375,3 +383,27 @@ func ValidateRoutingConfig(config api.RoutingConfig) fielderrors.ValidationError
 
 	return allErrs
 }
+
+func ValidateAPIServerExtendedArguments(config api.ExtendedArguments) fielderrors.ValidationErrorList {
+	return ValidateExtendedArguments(config, kapp.NewAPIServer().AddFlags)
+}
+
+func ValidateControllerExtendedArguments(config api.ExtendedArguments) fielderrors.ValidationErrorList {
+	return ValidateExtendedArguments(config, cmapp.NewCMServer().AddFlags)
+}
+
+func ValidateExtendedArguments(config api.ExtendedArguments, flagFunc func(*pflag.FlagSet)) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+
+	// check extended arguments for errors
+	for _, err := range cmdflags.Resolve(config, flagFunc) {
+		switch t := err.(type) {
+		case *fielderrors.ValidationError:
+			allErrs = append(allErrs, t)
+		default:
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("????", config, err.Error()))
```

Comment:
> does this include the entire config map in the message for every error of unknown type?

If we get an error type that we should never get and therefore, cannot relate to a single field, yes.